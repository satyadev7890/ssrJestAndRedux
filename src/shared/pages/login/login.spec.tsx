import * as React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './login';
import { RouteComponentProps } from "react-router-dom";

configure({ adapter: new Adapter() });

export interface Props extends RouteComponentProps {
    setSessionFlag: (val: boolean) => void;
    isSessionValid: boolean;
    history: any;
};

const props: Props  = {
    setSessionFlag: jest.fn(),
    isSessionValid: false,
    history: {push :  jest.fn() } 
}

describe('Login snapshot testing', () => {
    it('create snapshot with theme', () => {
      const login = shallow(<Login  {...props} />);
      expect(login).toMatchSnapshot();
    });
  });

describe('Login page testing', () => {
    let login: any = ''

    beforeEach(() => {
        login = shallow(<Login  {...props} />);
    });

    test('Login form validation with state change', () => {
        login.instance().setState({ userName: 'satya', passWord: 'password' });
        const submitBtn = login.find('#submit');
        submitBtn.simulate('click');
        expect(login.state().isInvalidFormDetails).toBeFalsy();
        expect(props.history.push).toHaveBeenCalled();
    })

    test('Login form validation with only username', () => {
        const jsdomAlert = window.alert;
        window.alert = () => {};
        const userName = login.find('#userName');
        userName.simulate('change', { target: { value: 'satya' } })
        const submitBtn = login.find('#submit');
        submitBtn.simulate('click');
        expect(login.state().isInvalidFormDetails).toBeTruthy();
        window.alert = jsdomAlert;
    })

    test('Login form validation with only password', () => {
        const jsdomAlert = window.alert;
        window.alert = () => {};
        const password = login.find('#passWord');
        password.simulate('change', { target: { value: 'password' } })
        const submitBtn = login.find('#submit');
        submitBtn.simulate('click');
        expect(login.state().isInvalidFormDetails).toBeTruthy();
        window.alert = jsdomAlert;
    })

    test('Login form validation with both username and password', () => {
        const userName = login.find('#userName');
        userName.simulate('change', { target: { value: 'satya' } })
        const password = login.find('#passWord');
        password.simulate('change', { target: { value: 'password' } })
        const submitBtn = login.find('#submit');
        submitBtn.simulate('click');
        expect(login.state().isInvalidFormDetails).toBeFalsy();
        expect(props.history.push).toHaveBeenCalled();
    })

});