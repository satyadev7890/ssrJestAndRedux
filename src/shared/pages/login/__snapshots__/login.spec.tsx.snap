// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot testing create snapshot with theme 1`] = `ShallowWrapper {}`;
